### PR TITLE
Support latest ProtocolLib

### LINKED ISSUE #386: HolographicDisplay Error
### Description
When spawning any MythicMobs mob from the default config spams this error.

### How to reproduce
Make a server and install following plugins
MythicMobs
Procollib
LibsDisguises
HolographicDisplays

1. Start the server
2. Type /mm mobs spawn SkeletalKnight
3. Wait for about 10-20 seconds for the error to appear

### Platform
- Holographic Displays version 2.4.5
- ProtocolLib version v4.6.0-SNAPSHOT-b491
- Spigot 1.16.4 R-0.1 SNAPSHOT

### Additional context
Error log:
https://pastebin.com/p0DU37Pu

### LINKED ISSUE #387: Error on join,leave, at random moment
### Description

HolographicDisplays] Unhandled exception number 16384 occured in onPacketSending(PacketEvent) for HolographicDisplays
java.lang.NoClassDefFoundError: com/comphenix/net/sf/cglib/proxy/Factory
at com.gmail.filoghost.holographicdisplays.bridge.protocollib.current.ProtocolLibHookImpl$1.onPacketSending(ProtocolLibHookImpl.java:87) ~[?:?]
at com.comphenix.protocol.injector.SortedPacketListenerList.invokeSendingListener(SortedPacketListenerList.java:195) ~[?:?]
at com.comphenix.protocol.injector.SortedPacketListenerList.invokePacketSending(SortedPacketListenerList.java:149) ~[?:?]
at com.comphenix.protocol.injector.PacketFilterManager.handlePacket(PacketFilterManager.java:535) ~[?:?]
at com.comphenix.protocol.injector.PacketFilterManager.invokePacketSending(PacketFilterManager.java:516) ~[?:?]
at com.comphenix.protocol.injector.netty.ProtocolInjector.packetQueued(ProtocolInjector.java:334) ~[?:?]
at com.comphenix.protocol.injector.netty.ProtocolInjector.onPacketSending(ProtocolInjector.java:296) ~[?:?]
at com.comphenix.protocol.injector.netty.ChannelInjector.processSending(ChannelInjector.java:434) ~[?:?]
at com.comphenix.protocol.injector.netty.ChannelInjector.access$800(ChannelInjector.java:71) ~[?:?]
at com.comphenix.protocol.injector.netty.ChannelInjector$3.handleScheduled(ChannelInjector.java:374) ~[?:?]
at com.comphenix.protocol.injector.netty.ChannelInjector$3.onMessageScheduled(ChannelInjector.java:344) ~[?:?]
at com.comphenix.protocol.injector.netty.ChannelProxy$2.schedulingRunnable(ChannelProxy.java:127) ~[?:?]
at com.comphenix.protocol.injector.netty.EventLoopProxy.execute(EventLoopProxy.java:95) ~[?:?]
at net.minecraft.server.v1_16_R3.NetworkManager.b(NetworkManager.java:293) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.NetworkManager.dispatchPacket(NetworkManager.java:249) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.NetworkManager.sendPacket(NetworkManager.java:225) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:1736) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.PlayerConnection.sendPacket(PlayerConnection.java:1709) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.EntityTrackerEntry.sendPlayerPacket(EntityTrackerEntry.java:49) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.EntityTrackerEntry.a(EntityTrackerEntry.java:201) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.EntityTrackerEntry.tick(EntityTrackerEntry.java:75) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.PlayerChunkMap.processTrackQueue(PlayerChunkMap.java:2052) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.PlayerChunkMap.g(PlayerChunkMap.java:2063) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.ChunkProviderServer.tickChunks(ChunkProviderServer.java:865) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.ChunkProviderServer.tick(ChunkProviderServer.java:743) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.WorldServer.doTick(WorldServer.java:460) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1351) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:377) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1204) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:992) ~[patched_1.16.4.jar:git-Paper-388]
at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:173) ~[patched_1.16.4.jar:git-Paper-388]
at java.lang.Thread.run(Thread.java:834) [?:?]

### How to reproduce

on player join
on player leave
or at random times

### Platform

- Server version (output of `/version`):  1.16.4 paper 388
- Holographic Displays version (check with `/hd`): 2.4.5 
- ProtocolLib version, if installed (check with `/protocol version`): #492
- Do you use plugins to allow players to join with multiple Minecraft versions (such as ViaVersion, ProtocolSupport, ...): viaversion

### LINKED ISSUE #388: HolographicDisplays & ProtocolLib Issue
### Description
https://hastebin.com/raw/suxinopari
Spams the console 3-4 times on player join, and every so often afterwards

### How to reproduce
Update from main release of ProtocolLib to the Latest Dev build.

### Platform

- Server version (output of `/version`): 1.16.4-R0.1-SNAPSHOT (Tuinity, fork of Paper)
- Holographic Displays version (check with `/hd`): 2.4.5
- ProtocolLib version, if installed (check with `/protocol version`): v4.6.0-SNAPSHOT-b492
- Do you use plugins to allow players to join with multiple Minecraft versions (such as ViaVersion, ProtocolSupport, ...): No

### Additional context
None

### PR DESCRIPTION
**Pull request by PimvanderLoos**
- Switched to JitPack for the ProtocolLib repository. See dmulloy2/ProtocolLib#1047.
- Using the new interface of generated classes when checking if a player is a proxy player. See dmulloy2/ProtocolLib#1053.

Closes #386, closes #387, closes #388

### CODE REVIEW

File: Plugin/src/main/java/com/gmail/filoghost/holographicdisplays/bridge/protocollib/current/ProtocolLibHookImpl.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public void onPacketSending(PacketEvent event) {
 					PacketContainer packet = event.getPacket();
 					Player player = event.getPlayer();
 					
-					if (player instanceof Factory) {
+					if (player instanceof ByteBuddyGenerated) {
```
Comment (dmulloy2) [inline Plugin/src/main/java/com/gmail/filoghost/holographicdisplays/bridge/protocollib/current/ProtocolLibHookImpl.java:87]: IMO it's not great to depend on an implementation detail. Would probably be better to use [event#isPlayerTemporary](https://github.com/dmulloy2/ProtocolLib/blob/master/src/main/java/com/comphenix/protocol/events/PacketEvent.java#L359)

File: Plugin/src/main/java/com/gmail/filoghost/holographicdisplays/bridge/protocollib/current/DebugHelper.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ private static String getPacketDescription(PacketContainer packetContainer) thro
 		// Get the first Minecraft super class
 		while (clazz != null && clazz != Object.class &&
 				(!MinecraftReflection.isMinecraftClass(clazz) || 
-				 Factory.class.isAssignableFrom(clazz))) {
+				 ByteBuddyGenerated.class.isAssignableFrom(clazz))) {
```
Comment (dmulloy2) [inline Plugin/src/main/java/com/gmail/filoghost/holographicdisplays/bridge/protocollib/current/DebugHelper.java:62]: Seems to be the same code as in [CommandPacket](https://github.com/dmulloy2/ProtocolLib/blob/master/src/main/java/com/comphenix/protocol/CommandPacket.java#L459), might be worth making that a public API?

File: Plugin/src/main/java/com/gmail/filoghost/holographicdisplays/bridge/protocollib/current/ProtocolLibHookImpl.java (hunk 1 above)
Comment (PimvanderLoos) [reply Plugin/src/main/java/com/gmail/filoghost/holographicdisplays/bridge/protocollib/current/ProtocolLibHookImpl.java:87]: Yeah, I definitely agree. Thanks for the suggestion!

File: Plugin/src/main/java/com/gmail/filoghost/holographicdisplays/bridge/protocollib/current/DebugHelper.java (hunk 2 above)
Comment (PimvanderLoos) [reply Plugin/src/main/java/com/gmail/filoghost/holographicdisplays/bridge/protocollib/current/DebugHelper.java:62]: Yeah, that'd be a bit cleaner. Thanks for the commit!

**Review by dmulloy2 — APPROVED** [review]

**Comment by strahil211** [comment]
Please I need this asap to stop the console spam error

**Comment by PimvanderLoos** [comment]
For anyone who would like to try this PR in its current state, I have published a temporary release [here](https://github.com/PimvanderLoos/HolographicDisplays/releases/tag/2.4.5).

**Comment by filoghost** [comment]
Thanks for the PR, and thanks to @dmulloy2 for adding the API.

I implemented it a bit differently: I added a dependency on 4.5.0 rather than the master branch (I couldn't add 4.4.0, as the build failed on Jitpack), and I didn't replace the old debug API yet (not available yet in 4.4.0). I also added a requirement for ProtocolLib 4.4+ in the startup. https://github.com/filoghost/HolographicDisplays/commit/0babfefe5ebe0353bd5cd9c4ddcd249d8069ad00

**Pull request closed without merging**